frontend/src/modules/3DViewer/settings/components/wellboreSelector.tsx

```diff
@@ -12,49 +15,122 @@ export type WellboreSelectorProps = {
 };
 
 export function WellboreSelector(props: WellboreSelectorProps): React.ReactNode {
+    const { onSelectedWellboreUuidsChange } = props;
+    const availableWellboreStatuses = Array.from(new Set(props.wellboreHeaders.map((header) => header.wellboreStatus)));
+    const availableWellborePurposes = Array.from(
+        new Set(props.wellboreHeaders.map((header) => header.wellborePurpose))
+    );
+    const [selectedWellboreStatuses, setSelectedWellboreStatuses] = useValidArrayState<string>({
+        initialState: availableWellboreStatuses,
+        validStateArray: availableWellboreStatuses,
+    });
+    const [selectedWellborePurposes, setSelectedWellborePurposes] = useValidArrayState<string>({
+        initialState: availableWellborePurposes,
+        validStateArray: availableWellborePurposes,
+    });
+    React.useEffect(() => {
+        onSelectedWellboreUuidsChange(
+            props.wellboreHeaders
+                .filter((header) => selectedWellborePurposes.includes(header.wellborePurpose))
+                .filter((header) => selectedWellboreStatuses.includes(header.wellboreStatus))
+                // .filter((header) => props.selectedWellboreUuids.includes(header.wellboreUuid))
```

Comment:
Can the two sequential `filter` be refactored into a reusable function, as it seems to be reused multiple times in the component. E.g. something providing an array `selectedWellBoreHeader`

... I'm assuming we can remove the third unused/commented `.filter` function here?

Reply:
Sure. This whole file is an intermediate state and requires some refactoring anyways. I created a common util function for now.